### PR TITLE
Deprecate across the board `max_satisfaction_weight`

### LINKED ISSUE #637: `max_satisfaction_weight` is only deprecated for some descriptors
Currently `max_satisfaction_weight` is deprecated only for:

- Generic `Descriptor`
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/mod.rs#L344-L346
- `Bare`
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/bare.rs#L86-L87
- `Sh`
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/sh.rs#L235-L237
- `Wsh`
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/segwitv0.rs#L104-L105
- `tr`
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/tr.rs#L312-L313

but not for:

- `Pkh`:
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/bare.rs#L250-L256
- `Wpkh`:
  https://github.com/rust-bitcoin/rust-miniscript/blob/469c113b7258742ba58bf23f9ed3cf8aed514ca8/src/descriptor/segwitv0.rs#L357-L363

Is this the intended behavior?
If not, I can easily (and gladly) open a PR to add deprecation notices to `Pkh` and `Wpkh`.

Cc @evanlinjin who originally deprecated these.
Related https://github.com/bitcoindevkit/bdk/issues/1036

### PR DESCRIPTION
**Pull request by storopoli**
Adds a `since=10.0.0` to all `max_satisfaction_weight` deprecations. Adds a note telling users to check #476 for more details.

Closes #637.

### CODE REVIEW

**Review by apoelstra — DISMISSED** [review]
ACK b656b8772e333ece5b338ef07343b90f2f19dbcb Thank you!!

**Review by tcharding — APPROVED** [review]
ACK b656b8772e333ece5b338ef07343b90f2f19dbcb

**Comment by tcharding** [comment]
Thanks for following through man.

**Comment by tcharding** [comment]
CI fails need you to use the new, non-deprecated versions, and there is some clippy stuff, could be unrelated to this PR but I didn't look.

**Comment by storopoli** [comment]
I'm on it...

**Comment by storopoli** [comment]
OK CI should pass now. I had to add some `dead_code` that were not being touched with deprecations anymore in some `Psbt` traits.

**Review by apoelstra — APPROVED** [review]
ACK 7fc76619abc33993e919e9e2c99d95f51f376cc9

**Comment by tcharding** [comment]
CI fail is unrelated to this PR.

miniscript has MSRV of 1.48 still, that won't work when its dependencies are using 1.56.1 - that should have been changed before last release.

**Comment by tcharding** [comment]
FWIW the most thorough way to go about this is, IMO, to fix the clippy as a separate PR, fix the MSRV as a separate PR, then rebase this. I understand if you don't want to do all that, I can try get to it, likely won't be till next week though.

**Comment by storopoli** [comment]
> FWIW the most thorough way to go about this is, IMO, to fix the clippy as a separate PR, fix the MSRV as a separate PR, then rebase this. I understand if you don't want to do all that, I can try get to it, likely won't be till next week though.

I was tinkering with trying to get things to work with Rust 1.48.0.
So I am free to make other PRs changing MSRV to 1.56.1? (Of course there are a bunch of documentation and "prose" related changes with that).
Good!

About the clippy warnings it seems also very easy.

Don't worry we can hold this until I finish these tasks and rebase.

**Comment by apoelstra** [comment]
> So I am free to make other PRs changing MSRV to 1.56.1?

Yep, go for it. Sorry that this hasn't been done already. I'm surprised we managed to update our `bitcoin` dependency without doing it.

**Comment by storopoli** [comment]
Ok PRs are up.

Suggested order of merging:

1. #639
2. #638 (this)
3. #640 

Of course I'll need to rebase once an ancestor PR is merged...

**Comment by apoelstra** [comment]
Honestly because these are independent (in terms of diff) and only affect the Github Actions CI (which we can't possibly bisect on) I'd be fine just merging them without them being rebased.

But see my comment on your first one.

**Review by tcharding — APPROVED** [review]
ACK 7fc76619abc33993e919e9e2c99d95f51f376cc9

**Pull request merged**
